homeassistant/components/conversation/default_agent.py

```diff
@@ -592,6 +613,89 @@ def _get_error_text(
         response_str = lang_intents.error_responses.get(response_key)
         return response_str or _DEFAULT_ERROR_TEXT
 
+    def register_trigger(
+        self,
+        sentences: list[str],
+        callback: TRIGGER_CALLBACK_TYPE,
+    ) -> core.CALLBACK_TYPE:
+        """Register a list of sentences that will trigger a callback when recognized."""
+        trigger_id = ulid()
+        self._trigger_sentences[trigger_id] = TriggerData(
+            sentences=sentences, callback=callback
+        )
+
+        # Force rebuild on next use
+        self._trigger_intents = None
+
+        unregister = functools.partial(self._unregister_trigger, trigger_id)
+        return unregister
+
+    def _rebuild_trigger_intents(self) -> None:
+        """Rebuild the HassIL intents object from the current trigger sentences."""
+        intents_dict = {
+            "language": self.hass.config.language,
+            "intents": {
+                trigger_id: {"data": [{"sentences": trigger_data.sentences}]}
+                for trigger_id, trigger_data in self._trigger_sentences.items()
+            },
+        }
+
+        self._trigger_intents = Intents.from_dict(intents_dict)
+        _LOGGER.debug("Rebuilt trigger intents: %s", intents_dict)
+
+    def _unregister_trigger(self, trigger_id: str) -> None:
+        """Unregister a set of trigger sentences."""
+        self._trigger_sentences.pop(trigger_id, None)
+
+        # Force rebuild on next use
+        self._trigger_intents = None
+
+    async def _match_triggers(self, sentence: str) -> ConversationResult | None:
+        """Try to match sentence against registered trigger sentences.
+
+        Calls the registered callbacks if there's a match and returns a positive
+        conversation result.
+        """
+        if not self._trigger_sentences:
+            # No triggers registered
+            return None
+
+        if self._trigger_intents is None:
+            # Need to rebuild intents before matching
+            self._rebuild_trigger_intents()
+
+        assert self._trigger_intents is not None
+
+        matched_triggers: set[str] = set()
+        speech: str | None = None
+        for result in recognize_all(sentence, self._trigger_intents):
+            trigger_id = result.intent.name
+            if trigger_id in matched_triggers:
+                # Already matched a sentence from this trigger
+                break
+
+            matched_triggers.add(trigger_id)
+            trigger_data = self._trigger_sentences[trigger_id]
+            if trigger_response := await trigger_data.callback(sentence):
+                speech = speech or trigger_response
```

Comment:
We should first fetch all callbacks and then execute them in parallel using `asyncio.gather`